### PR TITLE
acl: make auth method default across all types

### PR DESCRIPTION
**Pull request by pkazmierczak**
As discussed with the team, we want to have `default` auth method across all types, for better UX. 

Relates to https://github.com/hashicorp/nomad/issues/13120

### CODE REVIEW

File: command/login.go [hunk 1]
```diff
@@ -141,11 +140,14 @@ func (l *LoginCommand) Run(args []string) int {
 		for _, authMethod := range authMethodList {
 			if authMethod.Default {
 				l.authMethodName = authMethod.Name
+				if l.authMethodType == "" {
+					l.authMethodType = authMethod.Type
+				}
```
Comment (jrasell) [inline command/login.go:145]: What happens if the user specifies the auth method type, but this does not match the default one found? I assume the login will fail further down the flow, and wonder if we either want to error on this, or overwrite the passed flag value?

File: nomad/state/state_store_acl_sso.go [hunk 2]
```diff
@@ -186,10 +185,10 @@ func (s *StateStore) GetACLAuthMethodByName(ws memdb.WatchSet, authMethod string
 	return nil, nil
 }
 
-// GetDefaultACLAuthMethodByType returns a default ACL Auth Methods for a given
+// GetDefaultACLAuthMethod returns a default ACL Auth Methods for a given
 // auth type. Since we only want 1 default auth method per type, this function
 // is used during upserts to facilitate that check.
```
Comment (jrasell) [inline nomad/state/state_store_acl_sso.go:190]: I think we should update the function comment as I don't believe this is now correct based on the change of the logic.

**Review by jrasell — APPROVED** [review]
Looking good, thanks! I have noted a couple of minor things I think are worth changing and a question I am curious of your thoughts on.

File: command/login.go (hunk 1 above)
Comment (pkazmierczak) [reply command/login.go:145]: Right, it crossed my mind. I think it's fine to let it fail down the line, I don't think we should be overwriting things. Are you ok with this? or do you want to make a check here and fail quickly?

File: nomad/state/state_store_acl_sso.go (hunk 2 above)
Comment (pkazmierczak) [reply nomad/state/state_store_acl_sso.go]: resolved in [a56579e](https://github.com/hashicorp/nomad/pull/15869/commits/a56579ec97e83cd84f20ab0cdbbf93374a0af948)

File: .changelog/15869.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
+acl: made auth method default across all method types
+```
```
[the remark this answers is not in the document]
Comment (pkazmierczak) [reply .changelog/15869.txt]: resolved in [a56579e](https://github.com/hashicorp/nomad/pull/15869/commits/a56579ec97e83cd84f20ab0cdbbf93374a0af948)

File: command/login.go (hunk 1 above)
Comment (pkazmierczak) [reply command/login.go:145]: ok maybe it's best to fail quickly. 
[3e7dab8](https://github.com/hashicorp/nomad/pull/15869/commits/3e7dab826d82bb438deab46137b52b45059a2740)

**Pull request merged**